diff --git a/Library/Device/Nuvoton/M2351/Source/system_M2351.c b/Library/Device/Nuvoton/M2351/Source/system_M2351.c
index b0d6d04a..42e81d74 100644
--- a/Library/Device/Nuvoton/M2351/Source/system_M2351.c
+++ b/Library/Device/Nuvoton/M2351/Source/system_M2351.c
@@ -15,6 +15,7 @@
 #include <stdio.h>
 #include <stdint.h>
 #include "NuMicro.h"
+#include <inttypes.h>
 
 #if defined (__ARM_FEATURE_CMSE) &&  (__ARM_FEATURE_CMSE == 3U)
 #include "partition_M2351.h"
@@ -222,7 +223,7 @@ void SCU_IRQHandler(void)
             if(u32Reg & (1 << i))
             {
                 u32Addr = M32(SCU_BASE+info[i]+4);
-                printf("  %s(0x%08x) Alarm! illegal access by %s\n",ipname[i], u32Addr,master[M32(SCU_BASE+info[i])]);
+                printf("  %s(0x%08"PRIx32") Alarm! illegal access by %s\n",ipname[i], u32Addr,master[M32(SCU_BASE+info[i])]);
                 SCU->SVINTSTS = (1 << i);
                 break;
                 
